### PR TITLE
Use interface instead of os.File for possible broader use of the package

### PR DESCRIPTION
**Pull request by Trane9991**
Hey,
I wanted to use your package to parse some slow logs, however, I noticed, that it requires the `*os.File` as input params.

In my case, I have logs already in memory and I do not want to write them to file just to read them back. Also, using an interface is common practice for such operations.

Also, while testing the change I found other problems with tests that I fixed.

Thanks for the awesome package!

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/go-mysql?pullRequest=52) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/go-mysql?pullRequest=52) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/go-mysql?pullRequest=52) it.</sub>

**Comment by Trane9991** [comment]
@AlekSi any change to review this? Thanks!

File: Gopkg.toml [hunk 1]
```diff
@@ -6,3 +6,7 @@
 [[constraint]]
   name = "github.com/shirou/gopsutil"
   branch = "master"
+
+[[constraint]]
+  name = "github.com/mattetti/filebuffer"
```
Comment (percona-csalguero) [inline Gopkg.toml]: I think we don't need an extra dependency for a readk seeker in tests.

File: log/slow/parser_test.go [hunk 2]
```diff
@@ -1905,3 +1906,25 @@ func TestParseSlow026(t *testing.T) {
...
+	query := `
+# Time: 071218 11:48:27
+# User@Host: [SQL_SLAVE] @  []
+# Thread_id: 3  Schema: db1
+# Query_time: 0.000012  Lock_time: 0.000000  Rows_sent: 0  Rows_examined: 0
+use db2;
+SELECT fruit FROM trees;
+`
+
+	buf := filebuffer.New([]byte(query))
```
Comment (percona-csalguero) [inline log/slow/parser_test.go]: we can use `bytes.NewReader` here to avoid an extra dependency.

**Comment by Trane9991** [comment]
@percona-csalguero thanks. done!

**Comment by rnovikovP** [comment]
@Trane9991  Thanks 
Just, to speed up and make it  visible pls use    `re-request review` button 

image

And thanks for the contribution!

**Comment by Trane9991** [comment]
@rnovikovP thanks! missed it :)

**Comment by percona-csalguero** [comment]
Hi @Trane9991 . I am going to merge this after running some other tests. 
Thanks for the help

**Comment by Trane9991** [comment]
Thanks!

**Pull request merged**